packages/x-tree-view/src/TreeItem2/TreeItem2.test.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { createRenderer } from '@mui/internal-test-utils';
+import { TreeItem2 } from '@mui/x-tree-view/TreeItem2';
+import { treeItemClasses as classes } from '@mui/x-tree-view/TreeItem';
+import { TreeViewContext } from '@mui/x-tree-view/internals/TreeViewProvider/TreeViewContext';
+import { describeConformance } from 'test/utils/describeConformance';
+import { getFakeContextValue } from 'test/utils/tree-view/fakeContextValue';
+import { describeSlotsConformance } from 'test/utils/describeSlotsConformance';
+
+describe('<TreeItem2 />', () => {
+  const { render } = createRenderer();
+
+  describeConformance(<TreeItem2 itemId="one" label="one" />, () => ({
+    classes,
+    inheritComponent: 'li',
+    wrapMount: (mount) => (item: React.ReactNode) => {
+      const wrapper = mount(
+        <TreeViewContext.Provider value={getFakeContextValue()}>{item}</TreeViewContext.Provider>,
+      );
+      return wrapper.childAt(0);
+    },
+    render: (item) => {
+      return render(
+        <TreeViewContext.Provider value={getFakeContextValue()}>{item}</TreeViewContext.Provider>,
+      );
+    },
+    muiName: 'MuiTreeItem2',
+    refInstanceof: window.HTMLLIElement,
+    skip: ['reactTestRenderer', 'componentProp', 'componentsProp', 'themeVariants'],
+  }));
+
+  describeSlotsConformance({
```

Comment:
This works great for super basic slots that are always rendered, but `describeSlotsConformance` will need a lot more work to support more complex slots like `dragAndDropOverlay` or components like `DatePicker`.

But it's a 1st draft and for now I think it does the job.